sc-24sp/assignments/de02E/game-of-life/src/main.rs

```diff
@@ -0,0 +1,130 @@
+use core::time;
+use std::thread::sleep;
+
+use rand::Rng;
+
+#[derive(Clone,Copy,PartialEq)]
+enum CellState {
+    Alive,
+    Dead,
+    Warm,
+}
+
+pub struct Universe {
+    width: u32,
+    height: u32,
+    cells: Vec<CellState>,
+}
+
+impl Universe {
+    fn get_index(&self, row: u32, column: u32) -> usize {
+        (row * self.width + column) as usize
+    }
+    fn get_status(&self, index: usize) -> CellState {
+        self.cells[index]
+    }
+    fn live_neighbor_count(&self, row: u32, column: u32, state: CellState) -> u8 {
+        let mut neighbor_count: u8 = 0;
+        let mut y;
+        let mut x;
+        for i in 0..9 {
+                y = row;
+                x = column;
+                if i == 4 {
+                    continue;
+                }
+                if i/3 == 0 && y == 0 {
+                    y = &self.height-1;
+                } else if i/3 == 2 && y == &self.height-1 {
+                    y = 0;
+                } else {
+                    y = (y+(i/3)) - 1;
+                }
+                if i%3 == 0 && x == 0 {
+                    x = &self.width-1;
+                } else if i%3 == 2 && x == &self.width-1 {
+                    x = 0;
+                } else {
+                    x = (x+(i%3)) -1;
+                }
+                let status: &CellState = &self.cells[Self::get_index(&self, y,x)];
+                if (*status == state) {
+                    neighbor_count += 1;
+                }
+            }
+        neighbor_count
+        }
+    }
+
+fn main() {
+    let uniHeight = 35;
+    let uniWidth = 35;
+    let mut cell_vec: Vec<CellState> = Vec::new();
+    for i in 0..uniHeight*uniWidth {
+        let rng = rand::thread_rng().gen_range(0..3);
+        match rng {
+            0 => {
+                cell_vec.push(CellState::Dead);
+            }
+            1 => {
+                cell_vec.push(CellState::Alive);
+            }
+            2 => {
+                cell_vec.push(CellState::Warm);
+            }
+            _ => {
+                print!("Error!");
+            }
+        }
+    }
+    let mut universe = Universe{width: uniWidth,height: uniHeight,cells: cell_vec.clone()};
+    loop {
+        let mut properIndex = (0,0);
+        for i in 0..uniHeight {
+            for j in 0..uniWidth {
+                match universe.get_status(universe.get_index(i, j)) {
+                    CellState::Alive => {
+                        print!("███");
```

Comment:
is it three characters wide because of the aspect ratio of characters on the terminal?

Reply:
Yes.